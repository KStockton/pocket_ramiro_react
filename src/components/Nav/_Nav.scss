.nav {
  @include displayFlex(space-between, center, center, row);
  border-bottom: 1px #00000026 solid;
  background: rgb(2,0,36);
  background: linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(0,0,255,1) 0%, rgba(0,212,255,1) 82%);
}

.nav-wrapper {
  @include displayFlex(left, null, center, null);

  .nav-toggle {
    cursor: $nav-cursor;
    transform: translate(-50%, -50%);
    visibility: hidden;
  }
}

input[type=checkbox] + label {
  .bar {
    background-color: #000;
    display: block;
    height: 4px;
    transition: all 0.3s;
    margin: 3px;
    width: 36px;
    
    &:hover {
      cursor: $nav-cursor;
    }
  }
  .nav-top {
    transform: rotate(0);
    transform-origin: 18% 0%;
  }
  .nav-middle {
    opacity: 1;
  }
  .nav-bottom{
    transform: rotate(0);
    transform-origin: 20% 129%;
  }
}

input[type=checkbox]:checked + label {
  .bar {
    cursor: $nav-cursor;
  }
  .nav-top {
    transform: rotate(45deg);
    transform-origin: 18% 0%;
  }
  .nav-middle {
    opacity: 0;
  }
  .nav-bottom {
    transform: rotate(-45deg);
    transform-origin: 20% 129%;
  }
}

#menu {
  @include displayFlex(null, null, null, column);
  background: $nav-background-color;
  font-size: 20px;
  height: 100vh;
  padding: 15px;
  width: 93vw;

  a {
    @include displayFlex(null, null, center, null);
    text-decoration: none;
    color: white;
    border-bottom: 2px solid white;
    padding-bottom: 20px;
    padding-top: 15px; 
    font-size: 36px;
    
  }

  .menu-icons {
    padding-right: 19px;
    font-size: 40px;
  }
}

h3 span {
  font-weight: 400;
  font-size: 18px;
}

.nav-login {
  font-size: 18px;
  margin-right: 12px;
  text-decoration: none;
  color: #fff;

  &:hover {
    border: 2px solid black;
  }
}

.mobile-menu {
  position: absolute;
  z-index: 1;
}

