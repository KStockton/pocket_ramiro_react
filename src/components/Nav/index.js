import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class Nav extends Component {
  constructor() {
    super()
    this.state = {
      isSelected: false,
      title: ''
    }
  }

  handleSelected = (e) => {
    const { checked } = e.target
    this.setState({isSelected: checked})
  }

  handleClick = (event) => {
    if (!event.target.closest('.nav-link')) return;
    const { name } = event.target.closest('.nav-link')
    this.setState({title: name , isSelected: false })
  }

  render() {
    const {isSelected, title} = this.state

    const hamburgerNav = (
      <section id='menu' onClick={this.handleClick}>
        <NavLink to='/tickets' name='Tickets' className='nav-link'>
          <i className="material-icons menu-icons"> insert_chart</i>
            Tickets
        </NavLink>
        <NavLink to='/resources' name='resources' className='nav-link'>
          <i className="material-icons menu-icons"> insert_chart</i>
            Assets
        </NavLink>
        <NavLink to='/parts' name='Parts' className='nav-link'> 
          <i className="material-icons menu-icons"> business_center</i>
            Parts
        </NavLink>
        <NavLink to='/companies' name='Companies' className='nav-link'>
          <i className="material-icons menu-icons"> business </i>
          Companies
        </NavLink>
      </section>
    );

    return (
      <div>
        <div className='nav'>
        <section className='nav-wrapper'>
          <input
            id='hamburger'
            type='checkbox' 
            className='nav-toggle'
            onChange={this.handleSelected}
            checked={isSelected}
          />
          <label htmlFor='hamburger'>
            <div className='bar nav-top'></div>
            <div className='bar nav-middle'></div>
            <div className='bar nav-bottom'></div>
          </label>
        </section>
        {title.length === 0 ? <h3> <span>Pocket</span> Ramiro</h3> : <h3>{title}</h3>}
        <NavLink to="/login" className="nav-login">
          Log in
        </NavLink>
        </div>
        <section className="mobile-menu">
          {isSelected && hamburgerNav}
        </section>
      </div> 
    )
  }
};

export default Nav;
