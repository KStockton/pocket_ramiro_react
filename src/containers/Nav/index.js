import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteSession } from '../../thunks/deleteSession';
import { NavLink } from 'react-router-dom';

export class Nav extends Component {
  constructor() {
    super();
    this.state = {
      isSelected: false,
      title: ''
    };
  }

  handleSelected = (e) => {
    const { checked } = e.target;
    this.setState({isSelected: checked});
  }

  handleClick = (e) => {
    if (!e.target.closest('.nav-link')) {
      return;
    }
    const { name } = e.target.closest('.nav-link');
    this.setState({title: name, isSelected: false });
  }

  handleLogout = (e) => {
    this.props.deleteSession(this.props.session.user_id);
  }

  render() {
    const {isSelected, title} = this.state;
    const { name } = this.props.session;

    const hamburgerNav = (
      <section id='menu' onClick={this.handleClick}>
        <NavLink to='/tickets' name='Tickets' className='nav-link'>
          <i className="material-icons menu-icons"> insert_chart</i>
            Tickets
        </NavLink>
        <NavLink to='/resources' name='resources' className='nav-link'>
          <i className="material-icons menu-icons"> insert_chart</i>
            Resources
        </NavLink>
        <NavLink to='/parts' name='Parts' className='nav-link'> 
          <i className="material-icons menu-icons"> business_center</i>
            Parts
        </NavLink>
        <NavLink to='/resourcetypes' name='Resource-Types' className='nav-link'>
          <i className="material-icons menu-icons"> business </i>
          Resource-Types
        </NavLink>
      </section>
    );

    return (
      <div>
        <div className='nav'>
          <section className='nav-wrapper'>
            <input
              id='hamburger'
              type='checkbox' 
              className='nav-toggle'
              onChange={this.handleSelected}
              checked={isSelected}
            />
            <label htmlFor='hamburger'>
              <div className='bar nav-top'></div>
              <div className='bar nav-middle'></div>
              <div className='bar nav-bottom'></div>
            </label>
          </section>
          {title.length === 0 ? <h3> <span>Pocket</span> Ramiro</h3> : <h3>{title}</h3>}
          <div id='login-wrapper'>
            <NavLink to="/login" className="nav-login">
              { !this.props.session.user_id ? 
              <>
                <i className="material-icons">account_circle </i>
                <p id='logout'>Login</p> 
              </> 
                : 
              <>
                <i class="material-icons" id='login-icon'> how_to_reg</i>
                <p onClick={this.handleLogout } >Logout</p>
              </>
              }
            </NavLink>
          </div>
        </div>
        <section className="mobile-menu">
          {isSelected && hamburgerNav}
        </section>
      </div> 
    );
  }
}

export const mapStateToProps = state => ({
  session: state.session
});

export const mapDispatchToProps = dispatch => ({
  deleteSession: (id) => dispatch(deleteSession(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
