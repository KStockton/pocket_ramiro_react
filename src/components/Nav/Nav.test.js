import React from 'react';
import { shallow } from 'enzyme';
import Nav from './';

describe('Nav', () => {
  let wrapper 
  beforeEach(() => {
    wrapper = shallow(<Nav/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should render a Nav component if isSelected is true', () => {
    const isSelected = {isSelected: {isSelected: true } }
    wrapper.setState({isSelected})
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  })

});