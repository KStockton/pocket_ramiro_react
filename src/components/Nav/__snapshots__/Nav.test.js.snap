// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<div>
  <section
    className="nav-wrapper"
  >
    <input
      className="nav-toggle"
      id="hamburger"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="hamburger"
    >
      <div
        className="bar nav-top"
      />
      <div
        className="bar nav-middle"
      />
      <div
        className="bar nav-bottom"
      />
    </label>
  </section>
  <section />
</div>
`;

exports[`Nav should render a Nav component if isSelected is true 1`] = `
<div>
  <section
    className="nav-wrapper"
  >
    <input
      className="nav-toggle"
      id="hamburger"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="hamburger"
    >
      <div
        className="bar nav-top"
      />
      <div
        className="bar nav-middle"
      />
      <div
        className="bar nav-bottom"
      />
    </label>
  </section>
  <section>
    <section
      id="menu"
    >
      <ul>
        <li>
          Assets
        </li>
        <li>
          Parts and Inventory
        </li>
        <li>
          History
        </li>
      </ul>
    </section>
  </section>
</div>
`;
